### DISCUSSION TITLE
How to go about disabling "Getting Started" webviews?

### DISCUSSION THREAD

**Discussion by mrora** [eclipse-theia/theia#10593] [Q&A] [answered]
If someone wanted to disable webviews from being opened by plugins (e.g. a `Getting Started`/`What's New` page), how could this be done?

I read in another discussion (https://github.com/eclipse-theia/theia/discussions/9182) that this should be possible, and I am looking for some further guidance on how to go about that. Could someone point me in the right direction to get started on this?

Thanks!

**Comment by vince-fugnitto** [comment, marked as answer]
@mrora thank you for the discussion :+1: 

Trying to prevent welcome webviews from being opened by plugins is likely not trivial (as highlighted in #9182) since there is no clear way to distinguish if a webview is in fact a welcome view such as a "getting started" or "what's new" page versus a regular webview. In order not to prevent all webviews from being opened which would be a major functionality loss for your users, you'll need to find a way to be able to identify if a webview is in fact a welcome view (perhaps by substring title but this isn't ideal either).

Given that you are able to identify a webview is a welcome-view, you'll need to create a [custom theia extension](https://theia-ide.org/docs/authoring_extensions/) which re-implements logic to prevent the opening of such views for plugins.

You may want to look at the following:
- https://github.com/eclipse-theia/theia/blob/master/packages/plugin-ext/src/main/browser/webviews-main.ts
- https://github.com/eclipse-theia/theia/blob/master/packages/plugin-ext/src/main/browser/webview/webview-widget-factory.ts

**Reply by mrora** [reply]
thanks, @vince-fugnitto! looking through the code some more, i see what you mean. since webviews can be used for many different purposes, the chance of breaking functionality for other plugins makes this seem like more trouble than it may be worth. but, i now have a better idea of what would need to done in order to do that. 👍
